2_process/calc_doy_means.py

```diff
@@ -23,8 +23,7 @@ def main(out_file, in_file, lake_id):
     calc_doy_means(df, lake_id, out_file)
 
 if __name__ == '__main__':
-    lake_ids = ["120020150", "107072210"]
-    for lake_id in lake_ids:
-        out_file = f"2_process/out/doy_{lake_id}.csv"
-        in_file = f"1_fetch/out/tmp/pgdl_nhdhr_{lake_id}_temperatures.csv"
-        main(out_file, in_file, lake_id)
+    out_file = str(snakemake.output)
+    in_file = str(snakemake.input)
```

Comment:
Nice - I wouldn't have thought of this solution. It makes sense though - `snakemake.output` is an object that ultimately inherits from the Python class 'list', so `str()` behaves on it in the same way as on a standard Python list. Incidentally, that means that you can index `snakemake.output` just like a list, so another solution is `out_file = snakemake.output[0]`. The preferred solution, though, is to use named variables for the sake of clarity. You can give the variables names in the Snakefile.

Reply:
Makes sense—so is it fair to say that rule parameters should always be named, except for the script field and when there are many outputs like in the unzip step?

Reply:
Pretty much - I'd say that names for inputs, outputs, and params should be specified whenever possible.